dev/src/document.ts

```diff
@@ -23,9 +23,90 @@ import is from 'is';
 import {FieldPath} from './path';
 import {FieldTransform} from './field-value';
 import {Timestamp} from './timestamp';
-import {validatePkg} from './validate';
 import {isPlainObject} from './serializer';
 
+/**
+ * Returns a builder for DocumentSnapshot and QueryDocumentSnapshot instances.
+ * Invoke `.build()' to assemble the final snapshot.
+ *
+ * @private
+ * @class
+ */
+class DocumentSnapshotBuilder {
+  ref;
+  fieldsProto;
+  readTime;
+  createTime;
+  updateTime;
+  /**
+   * @private
+   * @hideconstructor
+   *
+   * @param {DocumentSnapshot=} snapshot An optional snapshot to base this
+   * builder on.
+   */
+  constructor(snapshot?) {
+    snapshot = snapshot || {};
+
+    /**
+     * The reference to the document.
+     *
+     * @type {DocumentReference}
```

Comment:
I would remove the JSDoc type annotation.

Reply:
Sadly we can't :(  Right now we rely on these for generating the reference documentation.